### PR TITLE
[bitnami/harbor-exporter] Add VIB tests

### PR DESCRIPTION
**Pull request by corico44**
### Description of the change

The main objective of this PR is to publish our Bitnami Harbor-exporter container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### Additional information

The binary "harbor_exporter" cannot be tested as it requires other components:
```
I have no name!@fe2f9e048fc8:/opt/bitnami/harbor-exporter$ harbor_exporter --help
2023-04-17T16:45:20Z [ERROR] [/cmd/exporter/main.go:27]: Failed to parse database.conn_max_lifetime: time: invalid duration ""
2023-04-17T16:45:20Z [ERROR] [/cmd/exporter/main.go:32]: Failed to parse database.conn_max_idle_time: time: invalid duration ""
2023-04-17T16:45:20Z [INFO] [/common/dao/base.go:67]: Registering database: type-PostgreSQL host- port-0 database- sslmode-"disable"
2023-04-17T16:45:20Z [ERROR] [/common/utils/utils.go:108]: failed to connect to tcp://:0, retry after 2 seconds :dial tcp :0: connect: connection refused
2023-04-17T16:45:22Z [ERROR] [/common/utils/utils.go:108]: failed to connect to tcp://:0, retry after 2 seconds :dial tcp :0: connect: connection refused

```

### CODE REVIEW

File: .vib/harbor-exporter/goss/harbor-exporter.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+group:
+  harbor:
+    exists: true
+user:
+  harbor:
+    exists: true
+command:
+  # Ensure a set of directories exist and the non-root user has write privileges to them
+  check-directories-exist-with-user:
+    exec: ls -dl /etc/ssl/certs /etc/pki/tls/certs/ /var/log/jobs 2>/dev/null | grep "drwxrwxr-x.*harbor"
```
Comment (FraPazGal) [inline .vib/harbor-exporter/goss/harbor-exporter.yaml:10]: This check will succeed if just one dir exists and I think `/var/log/jobs` is always present, so the test will pass even if both `/etc/ssl/certs /etc/pki/tls/certs/` are not present, which is not what we want.

In this case, I suggest adding a separate [file](https://github.com/goss-org/goss/blob/master/docs/manual.md#file) check for `/var/log/jobs`.

File: .vib/harbor-exporter/goss/harbor-exporter.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    exec: ls /opt/bitnami/harbor-exporter/bin/swagger 2>/dev/null
+    exit-status: 2
+  # Ensure a set of directories exist and the non-root user has write privileges to them
+  check-directories-exist-with-user:
+    exec: ls -dl /etc/ssl/certs /etc/pki/tls/certs/ 2>/dev/null | grep "drwxrwxr-x.*harbor"
+    exit-status: 0
+  # Ensure permissions for Internal TLS
+  check-permissions-system-certs:
+    exec: ls -l /etc/pki/tls/certs/ca-bundle.crt /etc/pki/tls/certs/ca-bundle.trust.crt /etc/ssl/certs/ca-certificates.crt 2>/dev/null | grep ".*-rw-rw-r--"
+    exit-status: 0
```
Comment (migruiz4) [inline .vib/harbor-exporter/goss/harbor-exporter.yaml:25]: Why use 'ls' and 'grep' instead of a goss `file` test?

Comment (corico44) [reply .vib/harbor-exporter/goss/harbor-exporter.yaml:25]: We don't know which of the files exists or not, so we cannot perform this check with the format you propose. So the use of `file` isn't possible in this case.

File: .vib/harbor-exporter/goss/harbor-exporter.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  /var/log/jobs:
+    exists: true
+    mode: "0775"
+    owner: harbor
+    filetype: directory
+command:
+  # Ensure swagger binary doesn't exist
+  check-swagger-binary:
+    exec: ls /opt/bitnami/harbor-exporter/bin/swagger 2>/dev/null
+    exit-status: 2
```
Comment (FraPazGal) [inline .vib/harbor-exporter/goss/harbor-exporter.yaml:17]: As done for https://github.com/bitnami/containers/pull/30739, I would modify this check to use Goss' `file` directive.

**Review by FraPazGal — APPROVED** [review]
LGTM!

**Pull request merged**